### ISSUE TITLE
Help with understanding the generated image

### ISSUE THREAD

**Issue by ajaykmr2292** [junyanz/pytorch-CycleGAN-and-pix2pix#428]
Hi,
  I trained the model for coloring b/w images. But the generated image looks like this:
![generated_image](https://user-images.githubusercontent.com/29298860/48111114-47f35780-e293-11e8-9776-928c15fc647c.png) 
and loss curve looks like this: 
![loss_curves](https://user-images.githubusercontent.com/29298860/48111189-b7694700-e293-11e8-857f-708a1a641ace.png)

Can you tell me what's the reason the generator is having so many small color patches (like too many small color patches in the image)? The input is just a black and white sketch.

I ran it for 500 epochs and the the model is performing really well on training set. So, but the output for test set is like the one shown above. So, can we conclude the model is overfit? 

Since the gen loss curve seems to saturate after 200 epochs, I tried to use early stopping (at 200 epochs assuming that 500 epochs was the overfitting case) and the test result is almost the same as the one above. Can you explain whats the reason for this behaviour and also can you suggest few things to try?
@junyanz

**Comment by junyanz** [comment]
You might be overfitting your training set. How many training images do you have? For colorization, it at least needs tens of thousands of images. If you don't have many training images, you should apply data argumentation (e.g., like cropping with `loadSize 286` and `fineSize 256`).

**Comment by ajaykmr2292** [comment]
I am actually implementing this paper [cGAN-based Manga Colorization](https://arxiv.org/pdf/1706.06918.pdf) which uses cgan but the author uses few training images for the purpose of colorization. 

Yes, I have just around 150 training images. But the author of the above-mentioned paper mentions that a single image is sufficient for the cgan model to learn a character (with crops though which I am doing using `resize_and_crop` parameter). That paper seems to be contradicting the general cGAN training procedure. So, can you please explain where I am going wrong.

Also, if cgan requires lots of images, can you suggest me a method to color a picture when I have a very small dataset?

**Comment by junyanz** [comment]
Their method has a few post-processing steps (as detailed in Sec 3.5). You can potentially their steps to remove noise in your results. I think pix2pix can overfit to 1-2 images as shown [here](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/416). 

I guess that the generalization ability does not only depend on the number of images,  but also on (1) the complexity of your dataset, and (2) if your test images look like your training images.
